# Applied ML Assignment 1
# Heart Attack Analaysis
## Introduction
 A heart attack occurs when an artery supplying your heart with blood and oxygen becomes blocked. A blood clot can form and block your arteries, causing a heart attack. This Heart Attack Analysis helps to understand the chance of attack occurrence in persons based on varied health conditions.

## Dataset
The dataset is Heart_Attack_Analysis_Data.csv. It has been uploaded to elearn. 
This dataset contains data about some hundreds of patients mentioning: Age, Sex, Exercise Include Angina(1=YES, 0=NO), CP_Type (Chest Pain)(Value 1: typical angina, Value2: atypical angina, Value 3: non-anginal pain, Value 4: asymptomatic), ECG Results, Blood Pressure, Cholesterol, Blood Sugar, Family History (Number of persons affected in the family), Maximum Heart Rate, Target -0=LESS CHANCE , 1= MORE CHANCE

## Aim
- Building a Predictive Model using Naïve Bayesian Approach (Which features decide heart attack?)
- Comment on the performance of this model using AUC-ROC, Precision, Recall, F_score, Accuracy

You need to 
1. Preprocess the data to enhance quality
2. Carry out descriptive summarization of data and make observations
3. Identify relevant, irrelevant attributes for building model. 
4. Use data visualization tools and make observations
5. Carry out the chosen analytic task. Show results including intermediate results, as needed
6. Evaluate the solution

Following are some points for you to take note of, while doing the assignment in Jupyter Notebook:
- State all your assumptions clearly
- List all intermediate steps and learnings
- Mention your observations/findings

## Submission:
### Importing necessary packages

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
from sklearn.preprocessing import RobustScaler, StandardScaler
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score, classification_report,recall_score,precision_score,f1_score,confusion_matrix,ConfusionMatrixDisplay,roc_curve,auc,precision_recall_curve
warnings.filterwarnings('ignore')

### Data Cleaning
Check the shape of the dataframe loaded into memory from the CSV file and see the datatypes used in the dataset given

In [55]:
df = pd.read_csv("C:/Users/akhil/Documents/Heart_Attack_Analysis_Data.csv")
df.head()
dfShape = df.shape
print("----------------------------------")
print("\nThe data frame has:\n")
print( str(dfShape[0]) + " rows")
print( str(dfShape[1]) + " columns")
print("----------------------------------")
print("\nWith following data types:\n")
df.info()
print("----------------------------------")
print("\nFirst 5 rows of data:\n")
df.head()

----------------------------------

The data frame has:

303 rows
11 columns
----------------------------------

With following data types:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype
---  ------          --------------  -----
 0   Age             303 non-null    int64
 1   Sex             303 non-null    int64
 2   CP_Type         303 non-null    int64
 3   BloodPressure   303 non-null    int64
 4   Cholestrol      303 non-null    int64
 5   BloodSugar      303 non-null    int64
 6   ECG             303 non-null    int64
 7   MaxHeartRate    303 non-null    int64
 8   ExerciseAngina  303 non-null    int64
 9   FamilyHistory   303 non-null    int64
 10  Target          303 non-null    int64
dtypes: int64(11)
memory usage: 26.2 KB
----------------------------------

First 5 rows of data:



Age  Sex  CP_Type  BloodPressure  Cholestrol  BloodSugar  ECG  \
0   63    1        3            145         233           1    0   
1   37    1        2            130         250           0    1   
2   41    0        1            130         204           0    0   
3   56    1        1            120         236           0    1   
4  -57    0        0            120         354           0    1   

   MaxHeartRate  ExerciseAngina  FamilyHistory  Target  
0           150               0              2       1  
1           187               0              1       1  
2           172               0              0       1  
3           178               0              1       1  
4           163               1              0       1

All columns seems to be of type `integer` and all columns have a value (303 not null), which means that all values are filled and there are no missing values.

Now to checking if any of the columns have data that is **negative valued**(Since values for columns such as age )

In [58]:
for col in df:
    print("Column " + col + " has negative values: " + str((df[col] < 0).any()) + "\n")
            

Column Age has negative values: True

Column Sex has negative values: False

Column CP_Type has negative values: False

Column BloodPressure has negative values: False

Column Cholestrol has negative values: False

Column BloodSugar has negative values: False

Column ECG has negative values: False

Column MaxHeartRate has negative values: False

Column ExerciseAngina has negative values: False

Column FamilyHistory has negative values: False

Column Target has negative values: False



**None** of the values seems to be negative in the dataset provided to us

# Hello
This is another normal paragraph
- this is a list
- this is another list item
